### PR TITLE
SDESK-2940 specify qcode type in vocabulary schema

### PR DESCRIPTION
**Pull request by mugurrus**

### CODE REVIEW

**Comment by tomaskikutis** [comment]
I'm not sure if you can simulate stuff on `react-test-renderer`. If you have a `onChange`  handler and it still doesn't work you can try this one https://stackoverflow.com/a/48197567/1175593

File: scripts/apps/vocabularies/tests/vocabularies.spec.js [hunk 1]
```diff
@@ -132,6 +137,29 @@ describe('vocabularies', () => {
...
+                const testRenderer = TestRenderer.create(<ItemsTableComponent ref={(ref) => component = ref}
+                    model={{name: null, qcode: null}}
+                    schema={schema}
+                    schemaFields={schemaFields}
+                    gettext={() => ''}
+                    remove={() => {}}
+                    update={update} />);
+
+                component.setState({items, itemsValidation});
+                ReactTestUtils.Simulate.change(testRenderer.root.findByProps({type: 'number'}), {target: {value: '2'}});
```
[the remark this answers is not in the document]
Comment (mugurrus) [reply scripts/apps/vocabularies/tests/vocabularies.spec.js]: The element returned by find is the qcode input, not ItemsTableComponent

[the remark this answers is not in the document]
Comment (tomaskikutis) [reply scripts/apps/vocabularies/tests/vocabularies.spec.js]: You're right. If you do get the element, but events still don't get simulated I'd try doing it with jasmine, check the link to StackOverflow I've posted.

**Comment by tomaskikutis** [comment]
I've fixed the test :) Merge the fix to this branch https://github.com/tomaskikutis/superdesk-client-core/tree/fix-qcode-test

**Pull request merged**